runtime/js/10_dispatch_buffer.js

```diff
@@ -0,0 +1,146 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+"use strict";
+
+((window) => {
+  const core = window.Deno.core;
+
+  function assert(cond) {
+    if (!cond) {
+      throw Error("assert");
+    }
+  }
+
+  ////////////////////////////////////////////////////////////////////////////////////////////
+  ////////////////////////////// General async handling //////////////////////////////////////
+  ////////////////////////////////////////////////////////////////////////////////////////////
+
+  // General Async response handling
+  let nextRequestId = 1;
+  const promiseTable = {};
+
+  function opAsync(opName, opRequestBuilder, opResultParser) {
+    // Make sure requests of this type are handled by the asyncHandler
+    // The asyncHandler's role is to call the "promiseTable[requestId]" function
+    core.setAsyncHandlerByName(opName, (bufUi8, _) => {
```

Comment:
Not entirely sure, but I have a feeling that using arrow function here might be less performant than having a named function definition before `opAsync`

Reply:
opResultParser is passed into this arrow function, so that makes it hard to extract this arrow function from this opAsync function

Reply:
a solution could be to call this function on an object that stores this opResultParser function, but I found this quickly makes the code a lot less readable. please let me know if you think this is useful (I could create such a solution and share it with you for review)

Reply:
@inteon is it required to pass that `opResultParser`? I think you could just directly call it in such function

Reply:
In #9457, instead of passing `bufferOpParseResult` as argument, I pass `jsonOpParseResult`. This way code can be shared between jsonOps and bufferOps.
https://github.com/inteon/deno/blob/2f9a9cabd0d7c48f440677ff21885eac30d22ce3/core/core.js#L320-L334
https://github.com/inteon/deno/blob/2f9a9cabd0d7c48f440677ff21885eac30d22ce3/core/core.js#L364-L378

Reply:
I see, let's land it as is then.